pkgs/build-support/dotnet/make-nuget-deps/default.nix

```diff
@@ -1,14 +1,35 @@
-{ linkFarmFromDrvs, fetchurl }:
+{ linkFarmFromDrvs, fetchurl, stdenvNoCC, zip }:
 { name, nugetDeps ? import sourceFile, sourceFile ? null }:
 linkFarmFromDrvs "${name}-nuget-deps" (nugetDeps {
   fetchNuGet = { pname, version, sha256 ? "", hash ? ""
     , url ? "https://www.nuget.org/api/v2/package/${pname}/${version}" }:
-    fetchurl {
+    stdenvNoCC.mkDerivation {
       name = "${pname}.${version}.nupkg";
-      # There is no need to verify whether both sha256 and hash are
-      # valid here, because nuget-to-nix does not generate a deps.nix
-      # containing both.
-      inherit url sha256 hash;
+      src = fetchurl {
+        name = "${pname}.${version}.nupkg";
+        # There is no need to verify whether both sha256 and hash are
+        # valid here, because nuget-to-nix does not generate a deps.nix
+        # containing both.
+        inherit url sha256 hash;
+      };
+      dontUnpack = true;
+      installPhase = ''
+        runHook preInstall
+        install -Dm644 "$src" "$out"
```

Comment:
I think we can omit `install` altogether with `--output-file` flag and use `runCommand` instead `stdenvNoCC.mkDerivation`. What do you think?

```nix
let
  src = fetchurl { … };
in
runCommand src.name
  {
    inherit src;
    nativeBuildInputs = [ zip ];
  }
  ''
    zip --temp-path "$TMPDIR" --output-file "$out" --delete "$src" .signature.p7s || (( $? == 12 ))
  ''
```

Reply:
Ah, never mind, just tried that and we still have to copy the archive if it does not contain a signature.

Reply:
Hah, we've come back to [an improved version of] where I started! I don't really mind either way but my poor computer has been building .NET all day and would like a rest.

Reply:
(Running an end-to-end test with that now.)

Reply:
Let's go with that.